keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/artifact/ImageHandler.kt

```diff
@@ -36,24 +37,32 @@ class ImageHandler(
           ArtifactCheckSkipped(artifact.type, artifact.name, "ActuationInProgress")
         )
       } else {
-        val latestVersion = artifact.findLatestArtifactVersion()
-        val latestBaseImageVersion = artifact.findLatestBaseAmiVersion()
+        val latestArtifactVersion = artifact.findLatestArtifactVersion()
+        val latestBaseAmiVersion = artifact.findLatestBaseAmiVersion()
         val image = artifact.findLatestAmi()
 
         if (image == null) {
           log.debug("No image found for {}", artifact.name)
         } else {
-          log.debug("Latest known {} version: {}", artifact.name, latestVersion)
+          log.debug("Latest known {} version: {}", artifact.name, latestArtifactVersion)
           log.debug("Latest baked {} version: {}", artifact.name, image.appVersion)
-          log.debug("Latest {} {} base AMI version: {}", artifact.vmOptions.baseOs, artifact.vmOptions.baseLabel, latestBaseImageVersion)
+          log.debug("Latest {} {} base AMI version: {}", artifact.vmOptions.baseOs, artifact.vmOptions.baseLabel, latestBaseAmiVersion)
           log.debug("Latest baked {} base AMI version: {}", artifact.name, image.baseAmiVersion)
         }
 
-        val versionsDiffer = image?.appVersion != latestVersion || image?.baseAmiVersion != latestBaseImageVersion
+        val versionsDiffer = image?.appVersion != latestArtifactVersion || image?.baseAmiVersion != latestBaseAmiVersion
         val regionsDiffer = !(image?.regions ?: emptySet()).containsAll(artifact.vmOptions.regions)
         if (image == null || versionsDiffer || regionsDiffer) {
-          log.info("Versions or regions for {} differ, launching bake", artifact.name)
-          launchBake(artifact, latestVersion)
+          if (bakeHistory.contains(latestArtifactVersion, latestBaseAmiVersion)) {
+            log.warn("Artifact version {} and base AMI version {} were baked previously", latestArtifactVersion, latestBaseAmiVersion)
+            publisher.publishEvent(RecurrentBakeDetected(latestArtifactVersion, latestBaseAmiVersion))
```

Comment:
Could you add an event listener into https://github.com/spinnaker/keel/blob/master/keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/telemetry/BakeryTelemetryListener.kt that increments a counter and tags by something like `artifactVersion+baseAmiVersion` (just one tag so it's easy to graph)?

Reply:
We probably want regions on there too